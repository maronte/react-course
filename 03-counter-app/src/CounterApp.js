import React, { useState } from "react";
import { PropTypes } from "prop-types";

const CounterApp = ({ value }) => {
    
    //Handle click event to increment the counter.
    const handleAdd = e => {
        /**
         * You can use the setState function to update the state.
         */
        setCounter(counter + 1);

        // Other way to update the state.
        // setCounter((counter) => counter + 1);
    };

    const handleSubtract = e => {
        setCounter(counter - 1);
    };

    const handleReset = e => {
        setCounter(value);
    }

    const [counter, setCounter] = useState(value);
    
    return (
        <>
            <h1>Counter App</h1>
            <h2>Counter: <span id="counterValue">{ counter }</span></h2>
            <button onClick={ handleAdd }>Increment</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubtract }>Decrement</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};

export default CounterApp;