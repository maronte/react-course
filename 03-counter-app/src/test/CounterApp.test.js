import { shallow } from "enzyme";
import CounterApp from "../CounterApp";
import "../setupTests";

describe("CounterApp", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CounterApp value={ 10 } />);
    });

    test("should shows the CounterApp correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should start counter with the given value", () => {
        const wrapper = shallow(<CounterApp value={ 0 } />);
        const expectedValue = "0";
        const actualValue = wrapper.find("#counterValue").text();
        expect(actualValue).toBe(expectedValue);
    });

    test("should increments the counter value when buttom to increment is pressed", () => {
        const expectedValue = "11";
        wrapper.find("button").at(0).simulate("click");
        const actualValue = wrapper.find("#counterValue").text();
        expect(actualValue).toBe(expectedValue);
    });
    test("should decrements the counter value when buttom to drecement is pressed", () => {
        const expectedValue = "9";
        wrapper.find("button").at(2).simulate("click");
        const actualValue = wrapper.find("#counterValue").text();
        expect(actualValue).toBe(expectedValue);
    });

    test("should reset the counter value when buttom to reset is pressed", () => {
        const expectedValue = "10";

        // First increment the counter.
        wrapper.find("button").at(0).simulate("click");

        wrapper.find("button").at(1).simulate("click");
        const actualValue = wrapper.find("#counterValue").text();
        expect(actualValue).toBe(expectedValue);
    });
});