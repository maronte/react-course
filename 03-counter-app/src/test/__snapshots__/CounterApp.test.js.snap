// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should shows the CounterApp correctly 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    Counter: 
    <span
      id="counterValue"
    >
      10
    </span>
  </h2>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</Fragment>
`;
